chainer/functions/connection/depthwise_convolution_2d.py

```diff
@@ -0,0 +1,183 @@
+import numpy
+
+from chainer import cuda
+from chainer import function
+from chainer.utils import conv
+from chainer.utils import type_check
+
+
+def _pair(x):
+    if hasattr(x, '__getitem__'):
+        return x
+    return x, x
+
+
+class DepthwiseConvolution2D(function.Function):
+
+    def __init__(self, stride=1, pad=0):
+        self.sy, self.sx = _pair(stride)
+        self.ph, self.pw = _pair(pad)
+
+    def check_type_forward(self, in_types):
+        n_in = in_types.size()
+        type_check.expect(2 <= n_in, n_in <= 3)
+
+        x_type = in_types[0]
+        w_type = in_types[1]
+        type_check.expect(
+            x_type.dtype.kind == 'f',
+            w_type.dtype.kind == 'f',
+            x_type.ndim == 4,
+            w_type.ndim == 4,
+            x_type.shape[1] == w_type.shape[1],
+        )
+
+        if n_in.eval() == 3:
+            b_type = in_types[2]
+            type_check.expect(
+                b_type.dtype == x_type.dtype,
+                b_type.ndim == 1,
+                b_type.shape[0] == w_type.shape[0] * w_type.shape[1],
+            )
+
+    def forward(self, inputs):
+        x, W = inputs[:2]
+        b = inputs[2] if len(inputs) == 3 else None
+        kh, kw = W.shape[2:]
+
+        xp = cuda.get_array_module(*x)
+        if xp is numpy:
+            self.col = conv.im2col_cpu(
+                x, kh, kw, self.sy, self.sx, self.ph, self.pw)
+        else:
+            self.col = conv.im2col_gpu(
+                x, kh, kw, self.sy, self.sx, self.ph, self.pw)
+
+        B, C, KY, KX, IY, IX = self.col.shape
+        D = W.shape[0]  # (D, C, KY, KX)
+        c_ = self.col.transpose(1, 0, 4, 5, 2, 3) \
+            .reshape((C, B * IY * IX, KY * KX))
+        w_ = W.transpose(1, 2, 3, 0).reshape((C, KY * KX, D))
```

Comment:
Current order is same as other convolutional function. (convolution_2d and dilated_convolution_2d).

Should I change it?